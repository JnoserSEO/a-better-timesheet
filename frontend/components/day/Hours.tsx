import React from 'react'
import FullWeekTupple, {DayEntry} from '../../models/fullweek.model'
import TimesheetView from '../../models/view.model'
import weekStyleFilter from '../../utility/week-style-filter'
import '../../styles/day/hours.scss'

interface HoursTdProps{
    day:DayEntry
    index:number;
}

const HoursTd:React.FC<HoursTdProps> =({day,index})=>{
    const hours = day.data.reduce((acc, curr) => acc + curr.hours, 0)
    return <td key={index} className="day-total">{`${hours} Hour${hours === 1 ? '' : 's'}`}</td> 
}

const Hours:React.FC<{days:FullWeekTupple,view:TimesheetView}> = ({days,view})=>{
    return <tr>{days.map((day, index) => weekStyleFilter(view,index,<HoursTd key={index} day={day} index={index} />) )}</tr>
}

export default Hours