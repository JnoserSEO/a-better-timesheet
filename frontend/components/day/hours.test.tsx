import React from 'react'
import { shallow } from 'enzyme'
import Hours from './Hours'
import { FullWeekTuple } from '../../models/fullweek.model'

beforeEach(() => {
    beforeEach(() => {
        Date.now = jest.fn(() => 1601401562455); // 2020-9-29
    });
});


const days: FullWeekTuple = [{ dayOfWeek: 'Sunday', date: new Date('2020-09-27T00:00:00.000Z'), data: [] },
{ dayOfWeek: 'Monday', date: new Date('2020-09-28T00:00:00.000Z'), data: [{ "hours": 4, "name": "Comp B" }] },
{ dayOfWeek: 'Tuesday', date: new Date('2020-09-29T00:00:00.000Z'), data: [{ "hours": 1, "name": "Comp A" }] },
{ dayOfWeek: 'Wednesday', date: new Date('2020-09-30T00:00:00.000Z'), data: [] },
{ dayOfWeek: 'Thursday', date: new Date('2020-10-01T00:00:00.000Z'), data: [] },
{ dayOfWeek: 'Friday', date: new Date('2020-10-02T00:00:00.000Z'), data: [] },
{ dayOfWeek: 'Saturday', date: new Date('2020-10-03T00:00:00.000Z'), data: [] }]


test('should render Hours correctly with no entries', () => {
    const wrapper = shallow(<Hours akey={3} days={days} view={'5 day'} />)
    expect(wrapper.find).toMatchSnapshot()
})


