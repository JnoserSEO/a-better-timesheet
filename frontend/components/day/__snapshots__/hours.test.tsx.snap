// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Hours correctly with no entries 1`] = `[Function]`;
